### PR TITLE
The schedule service: model, dao and first resource methods

### PR DESCRIPTION
**Pull request by ivannov**

### CODE REVIEW

File: microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.*;
+
+@ApplicationScoped
+public class ScheduleDAO {
+
+    private long sequence = 0L;
+
+    private final Map<Long, Schedule> scheduleMap = new HashMap<>();
```
Comment (karianna) [inline microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java]: application scoped bean, can it be access by multiple threads? Is this threadsafe?

Comment (ivannov) [inline microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java]: Application Scoped CDI beans are not thread safe by default (unlike EJBs). That is why I have put the code that mutates its shared state (mainly the sequence attribute) in synchronized block. Once I start adding new operations in the JAX-RS endpoint, I will look into the other DAO operations and make them thread safe too

File: microservice-schedule/pom.xml [hunk 2]
```diff
@@ -28,5 +28,59 @@
...
+                <dependency>
+                    <groupId>org.jboss.arquillian.container</groupId>
+                    <artifactId>arquillian-glassfish-embedded-3.1</artifactId>
+                    <version>1.0.0.Final</version>
+                    <scope>test</scope>
+                </dependency>
+                <dependency>
+                    <groupId>fish.payara.extras</groupId>
+                    <artifactId>payara-embedded-all</artifactId>
+                    <version>4.1.153</version>
```
Comment (smillidge) [inline microservice-schedule/pom.xml]: payara embedded dependency should be 4.1.1.163.0.1

**Comment by smillidge** [comment]
I tested on our pre-release microprofile jar and it worked fine. So ready to merge from my point of view.

**Comment by ivannov** [comment]
Very well :)

Let's wait for Martijn's comment on his concern and if he is fine, you may proceed to merging this PR. And I'll fix the payara-embedded-all version in a subsequent one?

**Comment by smillidge** [comment]
Yeah no problem on the version. I'll change the pom once it's merged. I'll also add building the Uberjar in a subsequent PR.

File: microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java (hunk 1 above)
Comment (karianna) [inline microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java]: OK, better to use thread safe data structures than synchronised blocks, but you can refactor that later

Comment (ivannov) [inline microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java]: Well, I didn't pay attention to exactly this part of the micro service. In a real world app the ID sequence is handled by the Persistence/Storage layer, not by the business logic. So I just wrote something that made sure some thread safety for generating IDs, without much paying attention what are the best practices of concurrent Longs.

I'll look how other microservices in this projects implemented it and will come up with something better in subsequent commits

**Comment by smillidge** [comment]
I'll merge so we can get the skeleton in and we can work on it from there

**Pull request merged**